app/services/katello/pulp3/repository/apt.rb

```diff
@@ -31,22 +39,20 @@ def sync_url_params(sync_options)
         end
 
         def mirror_remote_options
-          super.merge(
-            {
-              distributions: repo.deb_releases + "#{' default' unless repo.deb_releases.include? 'default'}"
-            }
-          )
+          if Setting['deb_enable_structured_apt']
+            distributions = repo.deb_pulp_distributions.join(' ')
+          else
+            distributions = 'default'
+          end
+
+          super.merge({distributions: distributions})
         end
 
         def publication_options(repository_version)
           ss = api.signing_services_api.list(name: SIGNING_SERVICE_NAME).results
           popts = super(repository_version)
-          popts.merge!(
-            {
-              structured: true, # publish real suites (e.g. 'stable')
-              simple: true # publish all into 'default'-suite
-            }
-          )
+          popts.merge!({ structured: true })
+          popts.merge!({ simple: true }) unless Setting['deb_enable_structured_apt']
```

Comment:
So we want to publish structured always and simple only if not enable_structured_apt?
Wasn't there an issue that we should only publish either simple or structured?

Reply:
We had an issue with publishing both, that led to us internally switching to publishing only one or the other as a hotfix, but that work around never made it into upstream. Eventually we fixed the underlying issue, so the workaround was no longer strictly necessary. The way it is now in this PR does not change the upstream behavior for when structured APT is disabled.